src/main/java/com/techiteasy/demo/services/CIModuleService.java

```diff
@@ -0,0 +1,42 @@
+package com.techiteasy.demo.services;
+
+import com.techiteasy.demo.dto.CIModuleDto;
+import com.techiteasy.demo.dto.CIModuleInputDto;
+import com.techiteasy.demo.mapping.CIModuleMapper;
+import com.techiteasy.demo.models.CIModule;
+import com.techiteasy.demo.repositories.CIModuleRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CIModuleService {
+
+    private final CIModuleRepository ciModuleRepository;
+
+    public CIModuleService(CIModuleRepository ciModuleRepository) {
+        this.ciModuleRepository = ciModuleRepository;
+    }
+
```

Comment:
Het is mooi om te zien dat je die in een functional programming geschreven hebt, het ziet netter uit.

Reply:
Dit is geen functional programming, maar het is een stream. Dit is gewoon onderdeel van java, maar je kunt dit ook prima oplossen met een simpele for-loop